packages/aila/src/core/chat/AilaStreamHandler.ts

```diff
@@ -44,7 +44,12 @@ export class AilaStreamHandler {
         await this.readFromStream();
       }
     } catch (e) {
-      this.handleStreamError(e);
+      try {
+        await this.handleStreamError(e);
+      } catch (error) {
+        this._chat.aila.errorReporter?.reportError(error);
```

Comment:
Beware that there is a ` this._chat.aila.errorReporter?.reportError(` within `handleStreamError`. It means it could double report an error if it's a non-helicone error object